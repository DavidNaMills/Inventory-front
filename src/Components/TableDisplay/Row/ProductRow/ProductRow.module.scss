.productRow{
    display: flex;
    align-items: center;
    
    &__infoCont{
        display: flex;
        flex-direction: column;
        margin-left: 20px;

    }
    
    

    &__cellCont{
        width: 100%;
        padding-top: 20px;
        display: flex;

        p{
            margin-right: 50px;
        }

        p:last-of-type {
            margin-right: none;
        }

    }

    &__cellCont_stack{
        width: 100%;
        padding-top: 10px;
        padding-bottom: 10px;
        display: flex;
        flex-direction: column;

        p{
            margin-right: 50px;
        }

        p:last-of-type {
            margin-right: none;
        }

    }



    span{
        font-weight: bold;
        margin-right: 10px;
    }
    
}
