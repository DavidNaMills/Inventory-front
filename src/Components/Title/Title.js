import React from 'react';
import classes from './title.module.scss';


const Title = (props) => {
    const { size = null, message = '', center = false } = props;
    let element = null;

    switch (size) {
        case 'h1':
            element = <h1 className={[classes.title_h1, center ? classes.title__center : null].join(' ')} >{message}</h1>
            break;
        case 'h2':
            element = <h2 className={[classes.title_h2, center ? classes.title__center : null].join(' ')}>{message}</h2>
            break;
        case 'h3':
            element = <h3 className={[classes.title_h3, center ? classes.title__center : null].join(' ')}>{message}</h3>
            break;
        case 'h4':
            element = <h4 className={[classes.title_h4, center ? classes.title__center : null].join(' ')}>{message}</h4>
            break;
        case 'h5':
            element = <h5 className={[classes.title_h5, center ? classes.title__center : null].join(' ')}>{message}</h5>
            break;
        case 'h6':
            element = <h6 className={[classes.title_h6, center ? classes.title__center : null].join(' ')}>{message}</h6>
            break;
        default:
            element = <h2 className={[classes.title_h2, center ? classes.title__center : null].join(' ')}>{message}</h2>
            break;
    };

    return (
        <div>
            {element}
        </div>
    )
}

export default React.memo(Title);