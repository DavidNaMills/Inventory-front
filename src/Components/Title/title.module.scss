@import '../../styles/variables/colors.module.scss';
@import '../../styles/fonts/fontSizes.module.scss';
@import '../../styles/mixins/title.module.scss';

.title {
    @include setTitle($h2, $title-color);
    
    &_h1{
        @include setTitle($h1, $title-color);
    };

    &_h2{
        @include setTitle($h2, $title-color);
    };
    
    &_h3{
        @include setTitle($h3, $title-color);
    }

    &_h4{
        @include setTitle($h4, $title-color);
    }

    &_h5{
        @include setTitle($h5, $title-color);
    }

    &_h6{
        @include setTitle($h6, $title-color);
    }

    &__center{
        text-align: center;
    }


}