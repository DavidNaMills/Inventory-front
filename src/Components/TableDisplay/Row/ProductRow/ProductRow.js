import React from 'react';
import { isMobileOnly } from 'react-device-detect';
import classes from './ProductRow.module.scss';
import ImageDisplay from '../../../ImageDisplay/ImageDisplay';
import Title from '../../../Title/Title';
import { useLocale } from '../../../../Context/LocaleContext';

const ProductRow = (props) => {
    const { config, data, withTitle=true, withImage=true, withInfo=true, stack=false } = props;
    const locale = useLocale();

    return (
        <div className={classes.productRow}>
            {withImage && <ImageDisplay url={data.url ? `/productImages/${data.url}` : null} alt={`${data.name}-${data.code}`} size='medium'/>}
            <div className={classes.productRow__infoCont}>
                {withTitle && <Title message={data[config.main[0]]} size='h4' />}
                {withInfo &&
                    <div className={stack? classes.productRow__cellCont_stack : classes.productRow__cellCont}>
                        <p><span>{locale[config.main[1]]}:</span>{data[config.main[1]]}</p>
                        <p><span>{locale[config.main[2]]}:</span>{data[config.main[2]]}</p>
                    </div>
                }
            </div>
        </div>
    )
}

export default ProductRow;