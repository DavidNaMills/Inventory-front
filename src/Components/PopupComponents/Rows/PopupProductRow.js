import React from 'react';
import { isMobileOnly } from 'react-device-detect';
import ProductRow from '../../TableDisplay/Row/ProductRow/ProductRow';
import classes from '../Popup.module.scss';
import Title from '../../Title/Title';
import {useLocale} from '../../../Context/LocaleContext';


const config2 = { main: ['', 'name', 'qtyInStock'] };

const PopupProductRow = (props) => {
    const { prods, isPortrait=true } = props;
    const locale = useLocale();

    const tempScrollableProd = {
        overflowY: 'auto',
        maxHeight: isPortrait ? '30vh' : '15vw',
    }

    return (
        prods.length > 0
            ? <div className={classes.popup__color2}>
                <div className={classes.popup__mediumSpacing}>
                    <Title message={`${locale['products_msg']}: ${prods.length>0 ? prods.length : 0} ${locale['in_ttl']}`} size='h4' />
                </div>
                <div style={tempScrollableProd}>
                    {
                        prods.map((c, d) =>
                            <div key={d} className={d % 2 ? classes.popup__row_odd : classes.popup__row_even}>
                                <ProductRow
                                    data={c}
                                    config={config2}
                                    withTitle={false}
                                    withImage={isMobileOnly ? false : true}
                                    stack={isMobileOnly ? true : false}
                                />
                            </div>
                        )
                    }
                </div>
            </div>
            : <p className={classes.popup__mediumSpacing}>{locale['no_products_type']}</p>
    )
}

export default PopupProductRow;