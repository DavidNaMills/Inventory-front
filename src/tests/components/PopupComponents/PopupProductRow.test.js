import React from 'react';
import '../../testHelpers/context';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';

import * as device from 'react-device-detect';

import PopupProductRow from '../../../Components/PopupComponents/Rows/PopupProductRow';
import Title from '../../../Components/Title/Title';
import ProductRow from '../../../Components/TableDisplay/Row/ProductRow/ProductRow';

const prods = [
    {name: 'cups', qtyInStock: 123},
    {name: 'bowls', qtyInStock: 12223},
];

describe('<PopupProductRow/> component test suite', ()=>{
    let wrapper;
    let orig;

    beforeEach(()=>{
        wrapper = shallow(<PopupProductRow prods={prods} isPortrait={true}/>);
        orig = device.isMobileOnly;
        device.isMobileOnly = false;
    });
    
    afterEach(()=>{
        device.isMobileOnly = orig;
    })

    it('matches snapshot', ()=>{
        const comp = renderer.create(<PopupProductRow prods={prods} isPortrait={false} />);
        const app = comp.toJSON();
        expect(app).toMatchSnapshot();
    });
    
    it('exists', ()=>{
        expect(wrapper.exists).toBeTruthy();
    });
    
    it('renders a <p> element if prods array is empty', ()=>{
        const tW = shallow(<PopupProductRow prods={[]} isPortrait={true}/>);
        const p = tW.find('p');
        expect(p.length).toBe(1);
        expect(p.props()).toHaveProperty('className', 'popup__mediumSpacing');
        expect(p.text()).toEqual('No products at this location');
    });
    
    it('renders a container <div> with classname "popup__color2"', ()=>{
        const div = wrapper.find('div');
        expect(div.length).toBe(5);
        expect(div.at(0).props()).toHaveProperty('className', 'popup__color2');
    });
    
    it('renders a <div> with classname "popup__mediumSpacing" and <Title/> component', ()=>{
        const div = wrapper.find('div');
        const ttl = div.at(1).find(Title);
        expect(div.at(1).props()).toHaveProperty('className', 'popup__mediumSpacing');
        expect(ttl.length).toBe(1);
        expect(ttl.props()).toHaveProperty('message', `Products: ${prods.length} in total`);
        expect(ttl.props()).toHaveProperty('size', `h4`);
    });
    
    it('renders a <div> with scrollable styling with styling for portrait as default', ()=>{
        const div = wrapper.find('div').at(2);
        expect(div.props().style).toHaveProperty('overflowY', 'auto');
        expect(div.props().style).toHaveProperty('maxHeight', '30vh');
    });
    
    it('renders a <div> with scrollable styling with styling for landscape', ()=>{
        const tW = shallow(<PopupProductRow prods={prods} isPortrait={false}/>);
        const div = tW.find('div').at(2);
        expect(div.props().style).toHaveProperty('overflowY', 'auto');
        expect(div.props().style).toHaveProperty('maxHeight', '25vw');
    });
    
    it('renders a <div> with odd and even classnames with a <ProductRow/> component as child', ()=>{
        const div = wrapper.find('div');
        expect(div.at(3).props()).toHaveProperty('className', 'popup__row_even');
        expect(div.at(3).find(ProductRow).length).toBe(1);
        expect(div.at(3).find(ProductRow).props()).toHaveProperty('withImage', true);
        expect(div.at(3).find(ProductRow).props()).toHaveProperty('stack', false);
        
        expect(div.at(4).props()).toHaveProperty('className', 'popup__row_odd');
        expect(div.at(4).find(ProductRow).length).toBe(1);
        expect(div.at(4).find(ProductRow).props()).toHaveProperty('withImage', true);
        expect(div.at(4).find(ProductRow).props()).toHaveProperty('stack', false);
    });
    
    it('renders a <div> with odd and even classnames with a <ProductRow/> component as child with settings for mobiles', ()=>{
        const tOrig = device.isMobileOnly;
        device.isMobileOnly = true;
        const tW = shallow(<PopupProductRow prods={prods} isPortrait={true}/>);
        const div = tW.find('div');
        expect(div.at(3).props()).toHaveProperty('className', 'popup__row_even');
        expect(div.at(3).find(ProductRow).length).toBe(1);
        expect(div.at(3).find(ProductRow).props()).toHaveProperty('withImage', false);
        expect(div.at(3).find(ProductRow).props()).toHaveProperty('stack', true);
        
        expect(div.at(4).props()).toHaveProperty('className', 'popup__row_odd');
        expect(div.at(4).find(ProductRow).length).toBe(1);
        expect(div.at(4).find(ProductRow).props()).toHaveProperty('withImage', false);
        expect(div.at(4).find(ProductRow).props()).toHaveProperty('stack', true);
        
        device.isMobileOnly = tOrig;
    });
});