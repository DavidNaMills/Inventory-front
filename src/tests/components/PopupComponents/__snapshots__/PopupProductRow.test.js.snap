// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PopupProductRow/> component test suite matches snapshot 1`] = `
<div
  className="popup__color2"
>
  <div
    className="popup__mediumSpacing"
  >
    <div>
      <h4
        className="title_h4 "
      >
        Products: 2 in total
      </h4>
    </div>
  </div>
  <div
    style={
      Object {
        "maxHeight": "25vw",
        "overflowY": "auto",
      }
    }
  >
    <div
      className="popup__row_even"
    >
      <div
        className="productRow"
      >
        <img
          alt="cups-undefined"
          className="imageDisplay imageDisplay__medium"
          src="noImage.jpg"
        />
        <div
          className="productRow__infoCont"
        >
          <div
            className="productRow__cellCont"
          >
            <p>
              <span>
                name
                :
              </span>
              cups
            </p>
            <p>
              <span>
                Qty
                :
              </span>
              123
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="popup__row_odd"
    >
      <div
        className="productRow"
      >
        <img
          alt="bowls-undefined"
          className="imageDisplay imageDisplay__medium"
          src="noImage.jpg"
        />
        <div
          className="productRow__infoCont"
        >
          <div
            className="productRow__cellCont"
          >
            <p>
              <span>
                name
                :
              </span>
              bowls
            </p>
            <p>
              <span>
                Qty
                :
              </span>
              12223
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
